src/main/java/org/folio/edge/patron/PatronHandler.java

```diff
@@ -307,13 +312,30 @@ private void setContentType(HttpServerResponse response, String contentType){
     }
   }
 
-  private static String updateRequestDateWithTimestamp(JsonObject requestMessage) {
+  private static String checkDates(JsonObject requestMessage) {
+    requestMessage = validateHoldsExpirationDate(requestMessage);
+    return updateRequestDateWithTimestamp(requestMessage);
+  }
 
+  private static String updateRequestDateWithTimestamp(JsonObject requestMessage) {
     final SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
     sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
     String timestamp = sdf.format(new Date());
 
-    requestMessage.put("requestDate", timestamp);
+    requestMessage.put(FIELD_REQUEST_DATE, timestamp);
     return requestMessage.encodePrettily();
   }
+
+  private static JsonObject validateHoldsExpirationDate(JsonObject requestMessage) {
+    String requestExpirationDate = requestMessage.getString(FIELD_EXPIRATION_DATE);
+    try {
+      if (requestExpirationDate != null && !requestExpirationDate.isEmpty()) {
+        new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ").parse(requestExpirationDate);
+      }
+    } catch (Exception parseEx) {
+      logger.debug("Exception parsing request expirationDate: " + requestExpirationDate);
+      requestMessage.remove(FIELD_EXPIRATION_DATE);
```

Comment:
The code makes me sad... 😢 I can't wait until we can remove it. Every EDS call from Chalmers will cause an exception and date format parser garbage. In the ideal world this would be a 422, not a silent removal, but I think it is fine as our short term solution.

Are we confident EDS is sending the expiration in this format when the field is present?

Reply:
I asked Ryan for a sample from and he can't find it in time. I tried digging up the logs and found a sample that looks like what I used to see from EDS.. so fingers crossed until that day comes.